apps/backend/src/app/api/v1/integrations/neon/projects/provision/route.tsx

```diff
@@ -0,0 +1,64 @@
+import { createApiKeySet } from "@/lib/api-keys";
+import { createProject } from "@/lib/projects";
+import { prismaClient } from "@/prisma-client";
+import { createSmartRouteHandler } from "@/route-handlers/smart-route-handler";
+import { neonAuthorizationHeaderSchema, projectDisplayNameSchema, yupNumber, yupObject, yupString, yupTuple } from "@stackframe/stack-shared/dist/schema-fields";
+import { decodeBase64 } from "@stackframe/stack-shared/dist/utils/bytes";
+
+export const POST = createSmartRouteHandler({
+  metadata: {
+    hidden: true,
+  },
+  request: yupObject({
+    body: yupObject({
+      display_name: projectDisplayNameSchema.defined(),
+    }).defined(),
+    headers: yupObject({
+      authorization: yupTuple([neonAuthorizationHeaderSchema.defined()]).defined(),
+    }).defined(),
+  }),
+  response: yupObject({
+    statusCode: yupNumber().oneOf([200]).defined(),
+    bodyType: yupString().oneOf(["json"]).defined(),
+    body: yupObject({
+      project_id: yupString().defined(),
+      super_secret_admin_key: yupString().defined(),
+    }).defined(),
+  }),
+  handler: async (req) => {
+    const { authorization } = req.headers;
```

Comment:
these lines appear multiple times in the codebase, maybe it should be a separate function.

Reply:
good catch